# Introduction
We all have once thought about building your own App. Here, we will do a comprehensive analysis of the Android app market by comparing thousands of apps in the Google Play store.
In this notebook, we will do a comprehensive analysis of the Android app market by comparing thousands of apps in the Google Play store.

we will compare thousands of apps in the Google Play Store so that we can gain insight into:

- How competitive different app categories (e.g., Games, Lifestyle, Weather) are

- Which app category offers compelling opportunities based on its popularity

- How many downloads you would give up by making your app paid vs. free

- How much you can reasonably charge for a paid app

- Which paid apps have had the highest revenue

- How many paid apps will recoup their development costs based on their sales revenue

# About the Dataset of Google Play Store Apps & Reviews

**Data Source:** <br>
App and review data was scraped from the Google Play Store by Lavanya Gupta in 2018. Original files listed [here](
https://www.kaggle.com/lava18/google-play-store-apps).

# Import Statements

In [ ]:
import pandas as pd

# Notebook Presentation

In [ ]:
# Show numeric output in decimal format e.g., 2.15
pd.options.display.float_format = '{:,.2f}'.format

# Read the Dataset

In [ ]:
df_apps = pd.read_csv('apps.csv')

# Data Cleaning

In [ ]:
df_apps.head()

App         Category  Rating  \
0                       Ak Parti Yardım Toplama           SOCIAL     NaN   
1                    Ain Arabic Kids Alif Ba ta           FAMILY     NaN   
2  Popsicle Launcher for Android P 9.0 launcher  PERSONALIZATION     NaN   
3                     Command & Conquer: Rivals           FAMILY     NaN   
4                                    CX Network         BUSINESS     NaN   

   Reviews  Size_MBs Installs  Type   Price Content_Rating           Genres  \
0        0      8.70        0  Paid  $13.99           Teen           Social   
1        0     33.00        0  Paid   $2.99       Everyone        Education   
2        0      5.50        0  Paid   $1.49       Everyone  Personalization   
3        0     19.00        0   NaN       0   Everyone 10+         Strategy   
4        0     10.00        0  Free       0       Everyone         Business   

     Last_Updated         Android_Ver  
0   July 28, 2017          4.1 and up  
1  April 15, 2016          3.0 and up  
2   July 11, 2018          4.2 and up  
3   June 28, 2018  Varies with device  
4  August 6, 2018          4.1 and up

In [ ]:
df_apps.shape

(10841, 12)

In [ ]:
df_apps.sample(5)

App     Category  Rating  Reviews  Size_MBs  \
4077                    ¡Ay Metro!         GAME    3.80      489     36.00   
9304  A+ Gallery - Photos & Videos  PHOTOGRAPHY    4.50   223941      6.90   
5148            Star Wars ™: DIRTY         GAME    4.50    38207     15.00   
9847                    Standoff 2         GAME    4.50   299046     47.00   
6006               Paint By Number       FAMILY    3.50      807      4.60   

        Installs  Type  Price Content_Rating        Genres      Last_Updated  \
4077      10,000  Free      0   Everyone 10+        Arcade    March 17, 2015   
9304  10,000,000  Free      0       Everyone   Photography    August 6, 2018   
5148     100,000  Paid  $9.99           Teen  Role Playing  October 19, 2015   
9847  10,000,000  Free      0     Mature 17+        Action     July 12, 2018   
6006     500,000  Free      0       Everyone        Casual       May 1, 2018   

             Android_Ver  
4077          4.0 and up  
9304  Varies with device  
5148          4.1 and up  
9847          4.1 and up  
6006          3.0 and up

In [ ]:
df_apps.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size_MBs', 'Installs', 'Type',
       'Price', 'Content_Rating', 'Genres', 'Last_Updated', 'Android_Ver'],
      dtype='object')

### Drop Unused Columns

We remove the columns called `Last_Updated` and `Android_Version` from the DataFrame in our analysis for now. We will not use these columns.

In [ ]:
df_apps.drop(['Last_Updated', 'Android_Ver'], axis=1, inplace=True)

In [ ]:
df_apps.head()

App         Category  Rating  \
0                       Ak Parti Yardım Toplama           SOCIAL     NaN   
1                    Ain Arabic Kids Alif Ba ta           FAMILY     NaN   
2  Popsicle Launcher for Android P 9.0 launcher  PERSONALIZATION     NaN   
3                     Command & Conquer: Rivals           FAMILY     NaN   
4                                    CX Network         BUSINESS     NaN   

   Reviews  Size_MBs Installs  Type   Price Content_Rating           Genres  
0        0      8.70        0  Paid  $13.99           Teen           Social  
1        0     33.00        0  Paid   $2.99       Everyone        Education  
2        0      5.50        0  Paid   $1.49       Everyone  Personalization  
3        0     19.00        0   NaN       0   Everyone 10+         Strategy  
4        0     10.00        0  Free       0       Everyone         Business

### Remove NaN values

In [ ]:
df_apps.isna()

App  Category  Rating  Reviews  Size_MBs  Installs   Type  Price  \
0      False     False    True    False     False     False  False  False   
1      False     False    True    False     False     False  False  False   
2      False     False    True    False     False     False  False  False   
3      False     False    True    False     False     False   True  False   
4      False     False    True    False     False     False  False  False   
...      ...       ...     ...      ...       ...       ...    ...    ...   
10836  False     False   False    False     False     False  False  False   
10837  False     False   False    False     False     False  False  False   
10838  False     False   False    False     False     False  False  False   
10839  False     False   False    False     False     False  False  False   
10840  False     False   False    False     False     False  False  False   

       Content_Rating  Genres  
0               False   False  
1               False   False  
2               False   False  
3               False   False  
4               False   False  
...               ...     ...  
10836           False   False  
10837           False   False  
10838           False   False  
10839           False   False  
10840           False   False  

[10841 rows x 10 columns]

In [ ]:
df_apps.isna().any()

App               False
Category          False
Rating             True
Reviews           False
Size_MBs          False
Installs          False
Type               True
Price             False
Content_Rating    False
Genres            False
dtype: bool

In [ ]:
df_apps.columns[df_apps.isna().any()].to_list()

['Rating', 'Type']

To find and remove the rows with the NaN values we can create a subset of the DataFrame based on where .isna() evaluates to True. \\
Analysis: We see that NaN values in ratings are associated with no reviews (and no installs). That makes sense.

In [ ]:
nan_rows = df_apps[df_apps['Rating'].isna()]
nan_rows.head()

App         Category  Rating  \
0                       Ak Parti Yardım Toplama           SOCIAL     NaN   
1                    Ain Arabic Kids Alif Ba ta           FAMILY     NaN   
2  Popsicle Launcher for Android P 9.0 launcher  PERSONALIZATION     NaN   
3                     Command & Conquer: Rivals           FAMILY     NaN   
4                                    CX Network         BUSINESS     NaN   

   Reviews  Size_MBs Installs  Type   Price Content_Rating           Genres  
0        0      8.70        0  Paid  $13.99           Teen           Social  
1        0     33.00        0  Paid   $2.99       Everyone        Education  
2        0      5.50        0  Paid   $1.49       Everyone  Personalization  
3        0     19.00        0   NaN       0   Everyone 10+         Strategy  
4        0     10.00        0  Free       0       Everyone         Business

In [ ]:
df_apps_clean = df_apps.dropna()
df_apps_clean.shape

(9367, 10)

### Find and Remove Duplicates


In [ ]:
duplicated_rows = df_apps_clean[df_apps_clean.duplicated()]
duplicated_rows.shape

(476, 10)

In [ ]:
duplicated_rows.head()

App Category  Rating  Reviews  \
946                       420 BZ Budeze Delivery  MEDICAL    5.00        2   
1133                                 MouseMingle   DATING    2.70        3   
1196  Cardiac diagnosis (heart rate, arrhythmia)  MEDICAL    4.40        8   
1231                                Sway Medical  MEDICAL    5.00        3   
1247              Chat Kids - Chat Room For Kids   DATING    4.70        6   

      Size_MBs Installs  Type   Price Content_Rating   Genres  
946      11.00      100  Free       0     Mature 17+  Medical  
1133      3.90      100  Free       0     Mature 17+   Dating  
1196      6.50      100  Paid  $12.99       Everyone  Medical  
1231     22.00      100  Free       0       Everyone  Medical  
1247      4.90      100  Free       0     Mature 17+   Dating

In [ ]:
df_apps_clean[df_apps_clean.App == '420 BZ Budeze Delivery']

App Category  Rating  Reviews  Size_MBs Installs  \
945  420 BZ Budeze Delivery  MEDICAL    5.00        2     11.00      100   
946  420 BZ Budeze Delivery  MEDICAL    5.00        2     11.00      100   

     Type Price Content_Rating   Genres  
945  Free     0     Mature 17+  Medical  
946  Free     0     Mature 17+  Medical

In [ ]:
df_apps_clean[df_apps_clean['App']== 'Instagram']

App Category  Rating   Reviews  Size_MBs       Installs  Type  \
10806  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   
10808  Instagram   SOCIAL    4.50  66577446      5.30  1,000,000,000  Free   
10809  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   
10810  Instagram   SOCIAL    4.50  66509917      5.30  1,000,000,000  Free   

      Price Content_Rating  Genres  
10806     0           Teen  Social  
10808     0           Teen  Social  
10809     0           Teen  Social  
10810     0           Teen  Social

**Analysis**: We see in above results of 'Instagram' App that 3 copies are retained having only chnage in Reviews. So if we want to remove redundant rows like this, we can provide a subset of column names as below. **The drop_duplicates() function only consider certain columns for identifying duplicates, by default use all of the columns.**

In [ ]:
df_apps_clean = df_apps_clean.drop_duplicates(subset=['App', 'Type', 'Price'])

In [ ]:
df_apps_clean.shape

(8199, 10)

In [ ]:
df_apps_clean[df_apps_clean['App']== 'Instagram']

App Category  Rating   Reviews  Size_MBs       Installs  Type  \
10806  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   

      Price Content_Rating  Genres  
10806     0           Teen  Social

# Find Highest Rated Apps

In [ ]:
df_apps_clean.sort_values(['Rating'], ascending=False).head()

App     Category  Rating  Reviews  Size_MBs Installs  \
21    KBA-EZ Health Guide      MEDICAL    5.00        4     25.00        1   
1230         Sway Medical      MEDICAL    5.00        3     22.00      100   
1227    AJ Men's Grooming    LIFESTYLE    5.00        2     22.00      100   
1224       FK Dedinje BGD       SPORTS    5.00       36      2.60      100   
1223      CB VIDEO VISION  PHOTOGRAPHY    5.00       13      2.60      100   

      Type Price Content_Rating       Genres  
21    Free     0       Everyone      Medical  
1230  Free     0       Everyone      Medical  
1227  Free     0       Everyone    Lifestyle  
1224  Free     0       Everyone       Sports  
1223  Free     0       Everyone  Photography

Only apps with very few reviews (and a low number on installs) have perfect 5 star ratings (most likely by friends and family). So we can not
rely exclusively on ratings alone to determine the quality of an app?

# Finding 5 Largest Apps in terms of Size (MBs)

In [ ]:
df_apps_clean.sort_values(['Size_MBs'], ascending=False).head()

App            Category  Rating  Reviews  \
9942   Talking Babsy Baby: Baby Games           LIFESTYLE    4.00   140995   
10687          Hungry Shark Evolution                GAME    4.50  6074334   
9943            Miami crime simulator                GAME    4.00   254518   
9944     Gangster Town: Vice District              FAMILY    4.30    65146   
3144                       Vi Trainer  HEALTH_AND_FITNESS    3.60      124   

       Size_MBs     Installs  Type Price Content_Rating  \
9942     100.00   10,000,000  Free     0       Everyone   
10687    100.00  100,000,000  Free     0           Teen   
9943     100.00   10,000,000  Free     0     Mature 17+   
9944     100.00   10,000,000  Free     0     Mature 17+   
3144     100.00        5,000  Free     0       Everyone   

                       Genres  
9942   Lifestyle;Pretend Play  
10687                  Arcade  
9943                   Action  
9944               Simulation  
3144         Health & Fitness

It seems to be an upper bound of 100 MB for the size of an app in Google play Store.

# Find the 5 App with Most Reviews

In [ ]:
df_apps_clean.sort_values(['Reviews'], ascending=False).head()

App       Category  Rating  \
10805                                  Facebook         SOCIAL    4.10   
10785                        WhatsApp Messenger  COMMUNICATION    4.40   
10806                                 Instagram         SOCIAL    4.50   
10784  Messenger – Text and Video Chat for Free  COMMUNICATION    4.00   
10650                            Clash of Clans           GAME    4.60   

        Reviews  Size_MBs       Installs  Type Price Content_Rating  \
10805  78158306      5.30  1,000,000,000  Free     0           Teen   
10785  69119316      3.50  1,000,000,000  Free     0       Everyone   
10806  66577313      5.30  1,000,000,000  Free     0           Teen   
10784  56642847      3.50  1,000,000,000  Free     0       Everyone   
10650  44891723     98.00    100,000,000  Free     0   Everyone 10+   

              Genres  
10805         Social  
10785  Communication  
10806         Social  
10784  Communication  
10650       Strategy

We find the most popular apps on the Android App Store. These include the usuals: Facebook, WhatsApp etc. And All are free, which is self-explantory.

# Visualise Categorical Data: Content Ratings

In [ ]:
ratings = df_apps_clean['Content_Rating'].value_counts()

In [ ]:
ratings

Everyone           6621
Teen                912
Mature 17+          357
Everyone 10+        305
Adults only 18+       3
Unrated               1
Name: Content_Rating, dtype: int64

In [ ]:
import plotly.express as px
import numpy as np

In [ ]:
fig = px.pie(labels=np.array(ratings.index),
values=ratings.values,
title="Content Rating",
names=ratings.index,
hole=0.6,
)
fig.update_traces(textposition='inside', textfont_size=15, textinfo='percent')

fig.show()


# Examining the Number of Installs

In [ ]:
df_apps_clean.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8199 entries, 21 to 10835
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             8199 non-null   object 
 1   Category        8199 non-null   object 
 2   Rating          8199 non-null   float64
 3   Reviews         8199 non-null   int64  
 4   Size_MBs        8199 non-null   float64
 5   Installs        8199 non-null   object 
 6   Type            8199 non-null   object 
 7   Price           8199 non-null   object 
 8   Content_Rating  8199 non-null   object 
 9   Genres          8199 non-null   object 
dtypes: float64(2), int64(1), object(7)
memory usage: 704.6+ KB


In [ ]:
df_apps_clean.Installs.describe()

count          8199
unique           19
top       1,000,000
freq           1417
Name: Installs, dtype: object

In [ ]:
df_apps_clean[['App', 'Installs']].groupby('Installs').count()

App
Installs           
1                 3
1,000           698
1,000,000      1417
1,000,000,000    20
10               69
10,000          988
10,000,000      933
100             303
100,000        1096
100,000,000     189
5                 9
5,000           425
5,000,000       607
50               56
50,000          457
50,000,000      202
500             199
500,000         504
500,000,000      24

In [ ]:
df_apps_clean.Installs = df_apps_clean.Installs.astype(str).str.replace(',', "")
df_apps_clean.Installs = pd.to_numeric(df_apps_clean.Installs)
df_apps_clean[['App', 'Installs']].groupby('Installs').count()

App
Installs        
1              3
5              9
10            69
50            56
100          303
500          199
1000         698
5000         425
10000        988
50000        457
100000      1096
500000       504
1000000     1417
5000000      607
10000000     933
50000000     202
100000000    189
500000000     24
1000000000    20

# Finding the Most Expensive Apps, Filter out the Junk, and Calculate a (ballpark) Sales Revenue Estimate

Let's examine the Price column more closely.


In [ ]:
df_apps_clean.Price.describe()

count     8199
unique      73
top          0
freq      7595
Name: Price, dtype: object

In [ ]:
df_apps_clean.Price = df_apps_clean.Price.astype(str).str.replace('$', "")
df_apps_clean.Price = pd.to_numeric(df_apps_clean.Price)
df_apps_clean.sort_values('Price', ascending=False).head(20)

<ipython-input-34-baa850548f3e>:1: FutureWarning:

The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.



App   Category  Rating  Reviews  Size_MBs  \
3946        I'm Rich - Trump Edition  LIFESTYLE    3.60      275      7.30   
2461              I AM RICH PRO PLUS    FINANCE    4.00       36     41.00   
4606               I Am Rich Premium    FINANCE    4.10     1867      4.70   
3145              I am rich(premium)    FINANCE    3.50      472      0.94   
3554                      💎 I'm rich  LIFESTYLE    3.80      718     26.00   
5765                       I am rich  LIFESTYLE    3.80     3547      1.80   
1946  I am rich (Most expensive app)    FINANCE    4.10      129      2.70   
2775                   I Am Rich Pro     FAMILY    4.40      201      2.70   
3221                  I am Rich Plus     FAMILY    4.00      856      8.70   
3114                       I am Rich    FINANCE    4.30      180      3.80   
1331          most expensive app (H)     FAMILY    4.30        6      1.50   
2394                      I am Rich!    FINANCE    3.80       93     22.00   
3897                       I Am Rich     FAMILY    3.60      217      4.90   
2193             I am extremely Rich  LIFESTYLE    2.90       41      2.90   
3856                   I am rich VIP  LIFESTYLE    3.80      411      2.60   
2281       Vargo Anesthesia Mega App    MEDICAL    4.60       92     32.00   
1407                    LTC AS Legal    MEDICAL    4.00        6      1.30   
2629                I am Rich Person  LIFESTYLE    4.20      134      1.80   
2481         A Manual of Acupuncture    MEDICAL    3.50      214     68.00   
4264         Golfshot Plus: Golf GPS     SPORTS    4.10     3387     25.00   

      Installs  Type  Price Content_Rating         Genres  
3946     10000  Paid 400.00       Everyone      Lifestyle  
2461      1000  Paid 399.99       Everyone        Finance  
4606     50000  Paid 399.99       Everyone        Finance  
3145      5000  Paid 399.99       Everyone        Finance  
3554     10000  Paid 399.99       Everyone      Lifestyle  
5765    100000  Paid 399.99       Everyone      Lifestyle  
1946      1000  Paid 399.99           Teen        Finance  
2775      5000  Paid 399.99       Everyone  Entertainment  
3221     10000  Paid 399.99       Everyone  Entertainment  
3114      5000  Paid 399.99       Everyone        Finance  
1331       100  Paid 399.99       Everyone  Entertainment  
2394      1000  Paid 399.99       Everyone        Finance  
3897     10000  Paid 389.99       Everyone  Entertainment  
2193      1000  Paid 379.99       Everyone      Lifestyle  
3856     10000  Paid 299.99       Everyone      Lifestyle  
2281      1000  Paid  79.99       Everyone        Medical  
1407       100  Paid  39.99       Everyone        Medical  
2629      1000  Paid  37.99       Everyone      Lifestyle  
2481      1000  Paid  33.99       Everyone        Medical  
4264     50000  Paid  29.99       Everyone         Sports

There are 15 "I am Rich" Apps in the Google Play Store apparently. They all cost 300 or more, which is the main point of the app.  This look dubious. So to keep our analyse safe and not effected by these side-effects, we put a cap of $250 in price.

### The most expensive apps sub $250

In [ ]:
df_apps_clean = df_apps_clean[df_apps_clean['Price'] < 250]
df_apps_clean.sort_values('Price', ascending=False).head(5)

App   Category  Rating  Reviews  Size_MBs  \
2281  Vargo Anesthesia Mega App    MEDICAL    4.60       92     32.00   
1407               LTC AS Legal    MEDICAL    4.00        6      1.30   
2629           I am Rich Person  LIFESTYLE    4.20      134      1.80   
2481    A Manual of Acupuncture    MEDICAL    3.50      214     68.00   
2463         PTA Content Master    MEDICAL    4.20       64     41.00   

      Installs  Type  Price Content_Rating     Genres  
2281      1000  Paid  79.99       Everyone    Medical  
1407       100  Paid  39.99       Everyone    Medical  
2629      1000  Paid  37.99       Everyone  Lifestyle  
2481      1000  Paid  33.99       Everyone    Medical  
2463      1000  Paid  29.99       Everyone    Medical

we look at the top 5 apps now, we see that 4 out of 5 are medical apps.

### Highest Grossing Paid Apps (ballpark estimate)

In [ ]:
df_apps_clean['Revenue_Estimate'] = df_apps_clean.Installs * df_apps_clean.Price
df_apps_clean.sort_values('Revenue_Estimate', ascending=False)[:10]

<ipython-input-36-bb5921b1a1d1>:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



App     Category  Rating  Reviews  Size_MBs  \
9220                      Minecraft       FAMILY    4.50  2376564     19.00   
8825                  Hitman Sniper         GAME    4.60   408292     29.00   
7151  Grand Theft Auto: San Andreas         GAME    4.40   348962     26.00   
7477            Facetune - For Free  PHOTOGRAPHY    4.40    49553     48.00   
7977        Sleep as Android Unlock    LIFESTYLE    4.50    23966      0.85   
6594            DraStic DS Emulator         GAME    4.60    87766     12.00   
6082                   Weather Live      WEATHER    4.50    76593      4.75   
7954                    Bloons TD 5       FAMILY    4.60   190086     94.00   
7633        Five Nights at Freddy's         GAME    4.60   100805     50.00   
6746     Card Wars - Adventure Time       FAMILY    4.30   129603     23.00   

      Installs  Type  Price Content_Rating                     Genres  \
9220  10000000  Paid   6.99   Everyone 10+  Arcade;Action & Adventure   
8825  10000000  Paid   0.99     Mature 17+                     Action   
7151   1000000  Paid   6.99     Mature 17+                     Action   
7477   1000000  Paid   5.99       Everyone                Photography   
7977   1000000  Paid   5.99       Everyone                  Lifestyle   
6594   1000000  Paid   4.99       Everyone                     Action   
6082    500000  Paid   5.99       Everyone                    Weather   
7954   1000000  Paid   2.99       Everyone                   Strategy   
7633   1000000  Paid   2.99           Teen                     Action   
6746   1000000  Paid   2.99   Everyone 10+    Card;Action & Adventure   

      Revenue_Estimate  
9220     69,900,000.00  
8825      9,900,000.00  
7151      6,990,000.00  
7477      5,990,000.00  
7977      5,990,000.00  
6594      4,990,000.00  
6082      2,995,000.00  
7954      2,990,000.00  
7633      2,990,000.00  
6746      2,990,000.00

The highest-grossing paid app Minecraft is at around $70 million. It’s quite interesting that Minecraft is listed in the Family category rather than in the Game category.

Analysing App Categories

If we were to release an app, should we go after a competitive category with many other apps? Or whould we target a popular category with a high number of downloads? Or perhaps you can target a category which is both popular but also one where the downloads are spread out among many different apps. That way, even if it’s more difficult to discover among all the other apps, your app has a better chance of getting installed, right? Let’s analyse this with bar charts and scatter plots and figure out which categories are dominating the market.

In [ ]:
df_apps_clean.Category.nunique()

33

In [ ]:
top10_category = df_apps_clean.Category.value_counts()[:10]
top10_category

FAMILY             1606
GAME                910
TOOLS               719
PRODUCTIVITY        301
PERSONALIZATION     298
LIFESTYLE           297
FINANCE             296
MEDICAL             292
PHOTOGRAPHY         263
BUSINESS            262
Name: Category, dtype: int64

### Highest Competition (Number of Apps)

In [ ]:
bar = px.bar(x = top10_category.index, # index = category name
             y = top10_category.values,
             title='Competition (Number of Apps per Category)')

bar.update_layout(xaxis_title='Category', yaxis_title='Number of Apps')
bar.show()

Most Popular Categories (Highest Downloads)

In [ ]:
category_installs = df_apps_clean.groupby('Category').agg({'Installs': pd.Series.sum})
category_installs.sort_values('Installs', ascending=True, inplace=True)

In [ ]:
h_bar = px.bar(x = category_installs.Installs,
               y = category_installs.index,
               orientation='h',
               title='Category Popularity (Number of Downloads)')

h_bar.update_layout(xaxis_title='Number of Downloads', yaxis_title='Category')
h_bar.show()

we see that Games and Tools are actually the most popular categories. If we plot the popularity of a category next to the number of apps in that category we can get an idea of how concentrated a category is. Do few apps have most of the downloads or are the downloads spread out over many apps?

### Category Concentration - Downloads vs. Competition

In [ ]:
cat_number = df_apps_clean.groupby('Category').agg({'App': pd.Series.count})

In [ ]:
cat_number

App
Category                 
ART_AND_DESIGN         61
AUTO_AND_VEHICLES      73
BEAUTY                 42
BOOKS_AND_REFERENCE   169
BUSINESS              262
COMICS                 54
COMMUNICATION         257
DATING                134
EDUCATION             118
ENTERTAINMENT         102
EVENTS                 45
FAMILY               1606
FINANCE               296
FOOD_AND_DRINK         94
GAME                  910
HEALTH_AND_FITNESS    243
HOUSE_AND_HOME         62
LIBRARIES_AND_DEMO     64
LIFESTYLE             297
MAPS_AND_NAVIGATION   118
MEDICAL               292
NEWS_AND_MAGAZINES    204
PARENTING              50
PERSONALIZATION       298
PHOTOGRAPHY           263
PRODUCTIVITY          301
SHOPPING              180
SOCIAL                203
SPORTS                260
TOOLS                 719
TRAVEL_AND_LOCAL      187
VIDEO_PLAYERS         148
WEATHER                72

In [ ]:
category_installs

Installs
Category                        
EVENTS                  15949410
BEAUTY                  26916200
PARENTING               31116110
MEDICAL                 39162676
COMICS                  44931100
LIBRARIES_AND_DEMO      52083000
AUTO_AND_VEHICLES       53129800
HOUSE_AND_HOME          97082000
ART_AND_DESIGN         114233100
DATING                 140912410
FOOD_AND_DRINK         211677750
EDUCATION              352852000
WEATHER                361096500
FINANCE                455249400
MAPS_AND_NAVIGATION    503267560
LIFESTYLE              503611120
BUSINESS               692018120
SPORTS                1096431465
HEALTH_AND_FITNESS    1134006220
SHOPPING              1400331540
PERSONALIZATION       1532352930
BOOKS_AND_REFERENCE   1665791655
ENTERTAINMENT         2113660000
NEWS_AND_MAGAZINES    2369110650
TRAVEL_AND_LOCAL      2894859300
VIDEO_PLAYERS         3916897200
FAMILY                4437554490
PHOTOGRAPHY           4649143130
SOCIAL                5487841475
PRODUCTIVITY          5788070180
TOOLS                 8099724500
COMMUNICATION        11039241530
GAME                 13858762717

In [ ]:
cat_merged_df = pd.merge(cat_number, category_installs, on='Category', how="inner")
print(f'The dimensions of the DataFrame are: {cat_merged_df.shape}')
cat_merged_df.sort_values('Installs', ascending=False)

The dimensions of the DataFrame are: (33, 2)


App     Installs
Category                              
GAME                  910  13858762717
COMMUNICATION         257  11039241530
TOOLS                 719   8099724500
PRODUCTIVITY          301   5788070180
SOCIAL                203   5487841475
PHOTOGRAPHY           263   4649143130
FAMILY               1606   4437554490
VIDEO_PLAYERS         148   3916897200
TRAVEL_AND_LOCAL      187   2894859300
NEWS_AND_MAGAZINES    204   2369110650
ENTERTAINMENT         102   2113660000
BOOKS_AND_REFERENCE   169   1665791655
PERSONALIZATION       298   1532352930
SHOPPING              180   1400331540
HEALTH_AND_FITNESS    243   1134006220
SPORTS                260   1096431465
BUSINESS              262    692018120
LIFESTYLE             297    503611120
MAPS_AND_NAVIGATION   118    503267560
FINANCE               296    455249400
WEATHER                72    361096500
EDUCATION             118    352852000
FOOD_AND_DRINK         94    211677750
DATING                134    140912410
ART_AND_DESIGN         61    114233100
HOUSE_AND_HOME         62     97082000
AUTO_AND_VEHICLES      73     53129800
LIBRARIES_AND_DEMO     64     52083000
COMICS                 54     44931100
MEDICAL               292     39162676
PARENTING              50     31116110
BEAUTY                 42     26916200
EVENTS                 45     15949410

In [ ]:
scatter = px.scatter(cat_merged_df, # data
                    x='App', # column name
                    y='Installs',
                    title='Category Concentration (Downloads vs Number of Apps)',
                    size='App',
                    hover_name=cat_merged_df.index,
                    color='Installs')

scatter.update_layout(xaxis_title="Number of Apps (Lower=More Concentrated)",
                      yaxis_title="Installs",
                      yaxis=dict(type='log'))

scatter.show()

we see is that the categories like Family, Tools, and Game have many different apps sharing a high number of downloads. But for the categories like video players and entertainment, all the downloads are concentrated in very few apps.

# Drill down to genres in each category

Let’s turn our attention to the Genres column. This is quite similar to the categories column but more granular.


In [ ]:
df_apps_clean.Genres.nunique()

114

In [ ]:
df_apps_clean.Genres.value_counts().sort_values(ascending=True)

Lifestyle;Pretend Play        1
Strategy;Education            1
Adventure;Education           1
Role Playing;Brain Games      1
Tools;Education               1
                           ... 
Personalization             298
Productivity                301
Education                   429
Entertainment               467
Tools                       718
Name: Genres, Length: 114, dtype: int64

In [ ]:
df_apps_clean.Genres.info()

<class 'pandas.core.series.Series'>
Int64Index: 8184 entries, 21 to 10835
Series name: Genres
Non-Null Count  Dtype 
--------------  ----- 
8184 non-null   object
dtypes: object(1)
memory usage: 127.9+ KB


In [ ]:
df_apps_clean['Genres'].str.split(';', expand=True)

0     1
21                     Medical  None
28                      Arcade  None
47                      Arcade  None
82                      Arcade  None
99                     Medical  None
...                        ...   ...
10824             Productivity  None
10828  Video Players & Editors  None
10829  Video Players & Editors  None
10831         News & Magazines  None
10835                   Arcade  None

[8184 rows x 2 columns]

In [ ]:
stack = df_apps_clean['Genres'].str.split(';', expand=True).stack()[50:75]
stack

357  0           Education
373  0              Arcade
378  0         Photography
382  0            Shopping
392  0              Puzzle
400  0     Personalization
402  0        Productivity
411  0              Sports
421  0         Photography
425  0             Medical
436  0              Puzzle
439  0    Health & Fitness
453  0           Education
     1           Education
462  0    Health & Fitness
467  0            Business
469  0           Education
479  0               Tools
482  0            Business
485  0             Medical
486  0               Tools
496  0     Personalization
504  0           Education
509  0           Education
521  0              Social
dtype: object

In [ ]:
num_genres = stack.value_counts()
num_genres

Education           6
Photography         2
Puzzle              2
Personalization     2
Medical             2
Health & Fitness    2
Business            2
Tools               2
Arcade              1
Shopping            1
Productivity        1
Sports              1
Social              1
dtype: int64

# Colour Scales in Plotly Charts - Competition in Genres


In [ ]:
bar = px.bar(
        x = num_genres.index[:15], # index = category name
        y = num_genres.values[:15], # count
        title='Top Genres (Sub-category)',
        hover_name=num_genres.index[:15],
        color=num_genres.values[:15],
        color_continuous_scale='Agsunset'
)

bar.update_layout(xaxis_title='Genre',
                  yaxis_title='Number of Apps',
                  coloraxis_showscale=False)

bar.show()



# Grouped Bar Charts: Free vs. Paid Apps per Category
Now that we’ve looked at the total number of apps per category and the total number of apps per genre, let’s see what the split is between free and paid apps.

In [ ]:
df_free_vs_paid = df_apps_clean.groupby(['Category', "Type"]).agg({'App':pd.Series.count})
df_free_vs_paid

App
Category          Type     
ART_AND_DESIGN    Free   58
                  Paid    3
AUTO_AND_VEHICLES Free   72
                  Paid    1
BEAUTY            Free   42
...                     ...
TRAVEL_AND_LOCAL  Paid    8
VIDEO_PLAYERS     Free  144
                  Paid    4
WEATHER           Free   65
                  Paid    7

[61 rows x 1 columns]

In [ ]:
df_free_vs_paid = df_apps_clean.groupby(["Category", "Type"], as_index=False).agg({'App': pd.Series.count})
df_free_vs_paid.head()

Category  Type  App
0     ART_AND_DESIGN  Free   58
1     ART_AND_DESIGN  Paid    3
2  AUTO_AND_VEHICLES  Free   72
3  AUTO_AND_VEHICLES  Paid    1
4             BEAUTY  Free   42

In [ ]:
g_bar = px.bar(df_free_vs_paid,
               x='Category',
               y='App',
               title='Free vs Paid Apps by Category',
               color='Type',
               barmode='group')

g_bar.update_layout(xaxis_title='Category',
                    yaxis_title='Number of Apps',
                    xaxis={'categoryorder':'total descending'},
                    yaxis=dict(type='log'))

g_bar.show()

What we see is that while there are very few paid apps on the Google Play Store, some categories have relatively more paid apps than others, including Personalization, Medical and Weather. So, depending on the category you are targeting, it might make sense to release a paid-for app

# Further Analysis:
But this leads to many more questions:

How much should you charge? What are other apps charging in that category?

How much revenue could you make?

And how many downloads are you potentially giving up because your app is paid?


### Approach:
To answer these, lets start with a box plot that shows the number of Installs for free versus paid apps. Then we can check the below points.
How does the median number of installations compare?
Is the difference large or small?

### Lost Downloads for Paid Apps

[Box Plots Guide](https://plotly.com/python/box-plots/) and the [.box API reference](https://plotly.com/python-api-reference/generated/plotly.express.box.html)



In [ ]:
box = px.box(df_apps_clean,
       y='Installs',
       x='Type',
       color='Type',
       notched=True,
       points='all',
      title='How Many Downloads are Paid Apps Giving Up?')

box.update_layout(yaxis=dict(type='log'))
box.show()

In the chart, we see that the median number of downloads for free apps is 500,000, while the median number of downloads for paid apps is around 5,000!
But does this mean we should give up on selling a paid app? Let’s see how much revenue we would estimate per category.

### Revenue by App Category
how much does the median app earn in the Tools category? If developing an Android app costs $30,000 or thereabouts, does the average photography app recoup its development costs?


In [ ]:
df_paid_apps = df_apps_clean[df_apps_clean['Type'] == 'Paid']

box = px. box(df_paid_apps,
              x= 'Category',
              y= 'Revenue_Estimate',
              title='How Much Can Paid Apps Earn?')

box.update_layout(xaxis_title='Category',
                  yaxis_title='Paid App Revenue',
                  xaxis={'categoryorder':'min ascending'},
                  yaxis=dict(type='log'))

box.add_hline(y=30000, line_width=1, line_color="purple")

box.show()

If an Android app costs dollar 30,000 (Red line) to develop, then the average app in very few categories would cover that development cost. The median paid photography app earned about $20,000. Many more app’s revenues were even lower - meaning they would need other sources of revenue like advertising or in-app purchases to make up for their development costs. However, certain app categories seem to contain a large number of outliers that have much higher (estimated) revenue - for example in Medical, Personalisation, Tools, Game, and Family.

# How Much Can we Charge? Paid App Pricing Strategies by Category.
So, if we were to list a paid app, how should we price it? To help us decide, we can look at how your competitors in the same category price their apps.

In [ ]:
df_paid_apps.Price.median()

2.99

In [ ]:
box = px.box(df_paid_apps,
             x='Category',
             y="Price",
             title='How Much Can we Charge?')

box.update_layout(xaxis_title='Category',
                  yaxis_title='Paid App Price',
                  xaxis={'categoryorder':'max descending'},
                  yaxis=dict(type='log'))

box.show()

Some categories have higher median prices than others.  The Medical apps have the most expensive apps as well as a median price of `$5.49`. In contrast, Personalisation apps are quite cheap on average at $1.49. Other categories which higher median prices are Business (`$4.99`) and `Dating` (`$6.99`). It seems like customers who shop in these categories are not so concerned about paying a bit extra for their apps.